### PR TITLE
Added GetInfo & ResizeVm APIs to enhance management capabilities

### PR DESCRIPTION
**Pull request by Leechael**

### CODE REVIEW

File: teepod/src/main_service.rs [hunk 1]
```diff
@@ -241,6 +241,48 @@ impl TeepodRpc for RpcHandler {
...
+            .manifest()
+            .context("failed to read manifest")?;
+        if let Some(vcpu) = request.vcpu {
+            manifest.vcpu = vcpu;
+        }
+        if let Some(memory) = request.memory {
+            manifest.memory = memory;
+        }
+        if let Some(disk_size) = request.disk_size {
+            manifest.disk_size = disk_size;
```
Comment (kvinwang) [inline teepod/src/main_service.rs:293]: Note: Resizing the disk size will not take effect

File: teepod/src/app.rs [hunk 2]
```diff
@@ -302,4 +302,51 @@ impl App {
...
+            }),
+            app_url: info.instance_id.as_ref().map(|id| {
+                format!(
+                    "https://{id}-{}.{}:{}",
+                    gw.tappd_port, gw.base_domain, gw.port
+                )
+            }),
+            app_id: info.manifest.app_id,
+            instance_id: info.instance_id,
+        })
```
Comment (kvinwang) [inline teepod/src/app.rs]: Looks like this should be extracted to a function that can be reused in `list_vms` and `get_vm`.

File: teepod/src/app.rs [hunk 3]
```diff
@@ -302,4 +302,51 @@ impl App {
             })
             .collect())
     }
+
+    pub fn get_vm(&self, id: &str) -> Option<pb::VmInfo> {
+        let state = self.state.lock().unwrap();
+        let vm = state.monitor.iter_vms().find(|vm| vm.info().manifest.id == id)?;
```
Comment (kvinwang) [inline teepod/src/app.rs]: The internal structure is a map, so linear searching is not required.

File: teepod/src/main_service.rs (hunk 1 above)
Comment (Leechael) [reply teepod/src/main_service.rs:293]: Let me add a comment regarding this situation and it's ok we only updating the manifest here for now.

**Review by kvinwang — APPROVED** [review]
LGTM

**Pull request merged**